File: LeetCode/java/1089_Duplicate_Zeros.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public void duplicateZeros(int[] arr) {
        List<Integer> list = new ArrayList<>();

        for (int i = 0; i < arr.length; i++) {
            if (list.size() == arr.length)
                break;

            if (arr[i] != 0) {
                list.add(arr[i]);
            } else {
                for (int j = 0; j < 2; j++) {
                    list.add(0);

                    if (list.size() == arr.length)
                        break;
                }
            }
        }

        for (int i = 0; i < list.size(); i++) {
            arr[i] = list.get(i);
        }
    }
}

File: LeetCode/cpp/1089_Duplicate_Zeros.cpp
class Solution {
public:
	void duplicateZeros(vector<int>& arr) {
		vector<int> v;

		for (int i = 0; i < arr.size(); i++) {
			if (v.size() == arr.size())
				break;

			if (arr[i] != 0) {
				v.push_back(arr[i]);
			}
			else {
				for (int j = 0; j < 2; j++) {
					v.push_back(0);

					if (v.size() == arr.size())
						break;
				}
			}
		}

		for (int i = 0; i < v.size(); i++) {
			arr[i] = v[i];
		}
	}
};